src/supabase/supabase_manager.py

```
@@ -587,6 +587,326 @@ def upload_all(self, tables: List[str] = None,
+            if csv_path.exists():
+                try:
+                    df = pd.read_csv(csv_path, low_memory=False)
+                    key_cols = [c for c in df.columns if c.endswith("_key")]
+                    for col in key_cols:
+                        null_count = int(df[col].isna().sum())
+                        if null_count > 0:
+                            null_key_warnings.append(
+                                f"{table_name}.{col}: {null_count} NULL values"
+                            )
+                except Exception:
+                    pass  # CSV read errors don't block validation
```

Ruff (0.15.2)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)